src/App/Utilities/TimerTask.cs

```diff
@@ -10,15 +10,23 @@ public class TimerTask
     {
         private readonly ILogger _logger;
         private readonly Action _action;
-        private readonly CancellationTokenSource _cancellationToken;
+        private readonly Func<Task> _func;
```

Comment:
I'd rename this to `_asyncAction` or `_actionTask` or something like that, func is not very descriptive IMHO